File: 3379-transformed-array/3379-transformed-array.py
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        res = []
        for i,n in enumerate(nums):
            indx = i + (n % len(nums))
            if indx >= len(nums):
                res.append(nums[indx%len(nums)])
            else:
                res.append(nums[indx])
        return res

File: 3379-transformed-array/3379-transformed-array.go
func constructTransformedArray(nums []int) []int {
	result := make([]int, len(nums))
	for i, v := range nums {
		if v > 0 {
			steps := (i + v) % len(nums)
			result[i] = nums[steps]
		} else if v < 0 {
			steps := ((i + v) % len(nums) + len(nums)) % len(nums)
			result[i] = nums[steps]
		} else {
			result[i] = v
		}
	}
	return result
}